feat(plugin-node-package): Add plugin-node-package and meta

infiniteluke opened the pull request:
<!-- What changes are being made? (What feature/bug is being fixed here?) -->
**What**:
Add some stuff

<!-- Why are these changes necessary? -->
**Why**:
So that we can share content between the emdaer website and the emdaer readme

<!-- How were these changes implemented? -->
**How**:
```md
<!--emdaer-p
  - '@emdaer/plugin-node-package'
  - path: '@emdaer/meta/lib/README/core-plugins.js'
    runEmdaer: true
-->
```

<!-- Have you done all of these things?  -->
**Checklist**:
<!-- add "N/A" to the end of each line that's irrelevant to your changes -->
<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->
- [X] Documentation
- [X] Tests
- [X] Ready to be merged <!-- In your opinion, is this ready to be merged as soon as it's reviewed? -->
- [X] Added myself to AUTHORS <!-- this is optional, see the contributing guidelines for instructions -->

<!-- Feel free to add additional comments -->

infiniteluke commented:
⚠️ Do not merge ⚠️ 
Needs another test

flipactual commented:
Works for me!

We'll clean this up for having content provided from other plugins when we further build out the Plugin architecture 👌

infiniteluke committed:
feat(plugin-node-package): Add plugin-node-package and meta
plugin-node-package requires a file and optionally executes it with provided arguments
meta exports README sections
This allows for the future emdaer website to share content with the main emdaer README in a composable way

infiniteluke committed:
chore: Merge master into plugin-package

infiniteluke committed:
docs: document how the emdaer README is made

infiniteluke committed:
test(plugin-node-package): test plugin-node-package and do minor cleanup

infiniteluke committed:
Merge branch 'master' into plugin-package
